integration-tests/tests/worker.rs

```diff
@@ -1693,3 +1693,78 @@ async fn resolve_components_from_name(deps: &EnvBasedTestDependencies, _tracing:
             ])
     );
 }
+
+#[test]
+#[tracing::instrument]
+#[timeout(120_000)]
+async fn promise(deps: &EnvBasedTestDependencies, _tracing: &Tracing) {
```

Comment:
Reason to move this test here is that the promise completion is going through the worker proxy, which is currently not supported in the worker executor test suite.

Optimally it would not go through the proxy if the worker is on the current instance, which would make the tests work there. But as this problem exists in other places in the code as well (e.g. worker fork), I would prefer to do it like this for now and make the proxy aware of workers on the current instance in a later change.